### PR TITLE
conan: fix build on darwin

### LINKED ISSUE #114490: conan 1.2.7 is not supported on darwin
**Description**
Attempting to install conan with nix-env on Mac OS X errors out with the following message:
```
error: Package ‘conan-1.27.0’ in /nix/store/90hd7lpqpy81a3zx179995k93sxyy91i-nixpkgs-21.05pre272691.04ac9dcd311/nixpkgs/pkgs/development/tools/build-managers/conan/default.nix:98 is not supported on ‘x86_64-darwin’, refusing to evaluate.
```

**To Reproduce**
1. `nix-env -i conan`

**Expected behavior**
As Mac OS X is a supported platform for conan, it's expected that it would install.

**Additional context**
This appears to be due to conan's dependency on the python3-distro package, which only compiles on Linux architectures.

**Notify maintainers**
@jonringer @drewrisinger @HaoZeke

**Metadata**
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS 10.15.7`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.6`
 - channels(tclemons): `"nixpkgs-21.05pre272691.04ac9dcd311"`
 - channels(root): `"nixpkgs-20.09pre213040.f77e057cda6"`
 - nixpkgs: `/Users/tclemons/.nix-defexpr/channels/nixpkgs`

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change

Fixes https://github.com/NixOS/nixpkgs/issues/114490
cc: @timclemons 
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/conan/default.nix [hunk 1]
```diff
@@ -98,6 +108,5 @@ in newPython.pkgs.buildPythonApplication rec {
     description = "Decentralized and portable C/C++ package manager";
     license = licenses.mit;
     maintainers = with maintainers; [ HaoZeke ];
-    platforms = platforms.linux;
```
Comment (Atemu) [inline pkgs/development/tools/build-managers/conan/default.nix:101]: `platforms = platforms.unix` would probably be appropriate, no?

File: pkgs/development/tools/build-managers/conan/default.nix [hunk 2]
```diff
@@ -20,6 +20,13 @@ let newPython = python3.override {
         inherit version;
         sha256 = "1vn1db2akw98ybnpns92qi11v94hydwp130s8753k6ikby95883j";
       };
+      patches = oldAttrs.patches or [] ++ [
+        # don't raise import error on non-linux os.
+        (fetchpatch {
+          url = "https://github.com/nir0s/distro/commit/25aa3f8c5934346dc838387fc081ce81baddeb95.patch";
+          sha256 = "0m09ldf75gacazh2kr04cifgsqfxg670vk4ypl62zv7fp3nyd5dc";
+        })
+      ];
```
Comment (Atemu) [inline pkgs/development/tools/build-managers/conan/default.nix:29]: Maybe add a comment about removing the patch one distro can be upgraded to >=1.2.0.

Should be possible from conan 1.29.0 onwards.

**Review by Atemu — APPROVED** [review]
Result of `nixpkgs-review pr 114533` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>conan</li>
  </ul>
</details>

Builds and executes.

File: pkgs/development/tools/build-managers/conan/default.nix (hunk 1 above)
Comment (marsam) [reply pkgs/development/tools/build-managers/conan/default.nix:101]: The code seems to only depend on python, so I think it's safe to assume that it runs on `python`'s platforms

File: pkgs/development/tools/build-managers/conan/default.nix (hunk 2 above)
Comment (marsam) [reply pkgs/development/tools/build-managers/conan/default.nix]: Fixed, thanks!

**Pull request merged**